### PR TITLE
Lib redesign skin fixes

### PR DESCRIPTION
**Pull request by ronso0**
**Edit** I removed the feature label from the sidebar in all skins. The feature title is available via BreadCrumbs above tracks table.

Tango:
- move BreadCrumbs back above tracks table
- style AutoDJ spinbox like beatsize spinboxes
- fix minor issues

Deere: move Lib maximize toggle above WButtonbar, push it next to Searchbox if sidebar is (almost) collapsed

2 1 1-skin-fixes-deere

LateNight:
- fix feature button bar style
- fix feature buttons & feature backgrounds
- style AutoDJ spinbox like beatsize spinboxes
- add translucent scrollbar handles to sidebar
- redesign toolbar divider to differ from splitter handle

2 1 1-skin-fixes_latenight

Please test!

### CODE REVIEW

**Comment by Be-ing** [comment]
screenshots?

**Comment by ronso0** [comment]
screens added

**Review by Be-ing — CHANGES_REQUESTED** [review]
The text above the control pane in LateNight is cut off for me with Qt5 and a scale factor of 2.5.
screenshot from 2018-05-19 17-08-58

Interestingly the feature name text above the track table does not show at all in Tango:
screenshot from 2018-05-19 17-18-01
While we have only one feature showing, I think this is preferable. If you could make the other skins do this, that would be nice.

Aside from that I noticed some minor code quality nitpicks.

File: res/skins/Deere/library.xml [hunk 1]
```diff
@@ -21,7 +21,11 @@
         <Children>
           <WidgetGroup>
             <Layout>vertical</Layout>
+            <Size>50min,-1me</Size>
             <Children>
+
+              <Template src="skin:library_sidebar_tester.xml"/>
```
[the remark this answers is not in the document]
Comment (ronso0) [reply res/skins/Deere/library.xml]: I can name it `library_sidebar_width_detector.xml`

[the remark this answers is not in the document]
Comment (ronso0) [reply res/skins/Deere/library.xml]: or we move the button bar outside the splitter, then this SizeAwareStack can be removed as well

**Comment by ronso0** [comment]
> The text above the control pane in LateNight is cut off for me with Qt5 and a scale factor of 2.5.

yep, I used a different hack to hide it. works @100% but scaling doesn't respect all sizes declared in css. Changed it so it works like in Tango now.

> While we have only one feature showing, I think this is preferable. If you could make the other skins do this, that would be nice.

Sure. Also applied some basic fixes in Shade.
@daschuer Do you plan to update the Shade library styling? Apart from the feature icons it looks like qt default.

**Comment by ronso0** [comment]
Done.
Please test

**Comment by Be-ing** [comment]
The jmigual-library-redesign branch has been merged with the latest updates from master. @ronso0 can you fix the merge conflicts with this branch again?

**Comment by ronso0** [comment]
I'm about to do so, but it's a bit messy now

**Comment by ronso0** [comment]
Though I'm wondering why the Library preview button in LateNight has some kind of minimum width defined: the preview column can be resized but the button itself occupies 65px anyway, so the icon is shifted and only partially visible if the preview column is too small

**Comment by nopeppermint** [comment]
in latenight if you disable "show text in sidebar icon" it doesn't rescale automatically, you need to change the window size for this..


latenight

the bpm "locked" is somehow broken:

bpm

**Comment by nopeppermint** [comment]
Shade Autodj Fade time is almost invisible
shade

**Comment by nopeppermint** [comment]
library styling for latenight is somehow broken:
latenight_hover compared to 2.1
latenight_2 1

- hover for preview button is extended over his colon and the played checkbox locks better in 2.1

**Comment by nopeppermint** [comment]
Deere preview button is broken
deere

**Comment by ronso0** [comment]
> Deere preview button is broken

Strange, didn't touch it. Fixed.

> I'm wondering why the Library preview button in LateNight has some kind of minimum width

Got it! It was a leftover/merge conflict:
```
WBaseLibrary QPushButton {
  min-width: 65px;
  min-height: 22px;
  max-height: 22px;
```

> in latenight if you disable "show text in sidebar icon" it doesn't rescale automatically

Can't confirm, behaves basicallly the same in all skins.
Though I can't test with most recent lib-redesign branch because I messd up my qt souces somehow..

> Shade Autodj Fade time is almost invisible

Those special library features are still unstyled as I was waiting for some feedback from @daschuer 
So it's the OS them issue again. I'll take care of that and the feature buttons soonish

**Comment by ronso0** [comment]
Now Shade library is styled as well.
Please test

**Comment by nopeppermint** [comment]
it seems that there are now "Conflicting files"

**Comment by nopeppermint** [comment]
latenight:
fixed:
- played checkbox

still open:
- hover for preview button is extended over his column

And I got a crash while I was trying to load a random track to autodj

```
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Warning [Main]: Could not load random track.
Segmentation fault
```
as Autodj source I had an empty crate, but I was not able to reproduce this crash..

**Comment by nopeppermint** [comment]
fixed:
- Deere preview button

> Though I can't test with most recent lib-redesign branch because I messd up my qt souces somehow..

I just compiled your branch, or should I compile the lib-redesign branch and copy only the skin folder from your branch?

**Comment by ronso0** [comment]
> it seems that there are now "Conflicting files"

Fixed..again. @Be-ing @daschuer Please review and merge this so I don't need to fix the merge conflicts again. Any further issues can be fixed in a follow-up PR

**Comment by ronso0** [comment]
> And I got a crash while I was trying to load a random track to autodj

Not related to this PR as it changes skin files only.
Please post those issues to lib-redesign branch or to respective zulip chat.

**Comment by ronso0** [comment]
@nopeppermint 
Due to the conflicts I just rebased to jmigual-lib-redesign, so it shouldn't make any difference.

**Comment by nopeppermint** [comment]
@ronso0  nice ! 
there might be more minor fix but in general it looks good on all skins under Linux Mint x64

one thing I see compared with branch 2.1, the title of the currently selected widget (library, autodj... ) needs a lot of vertical space.

compare_2 1_2 2

I think one's the suggested layout in https://github.com/mixxxdj/mixxx/pull/1117#issuecomment-333279581  is implemented, we need less space as well

> Single library after first start: search box next to bread crumbs (goes below if table width gets too small to show both side by side))

**Comment by Be-ing** [comment]
There are still some small issues, but those can be fixed later. This branch has enough improvements to be merged now. Thanks @ronso0, and thanks @nopeppermint for review.

**Comment by Be-ing** [comment]
Minor issues in Tango:
  * The cover art toggle icon feels out of place and IMO wastes space. Couldn't it be moved to the skin settings menu?
  * Although I can use the splitter between the control pane and track table, I don't see it so it would be difficult to discover if I was not looking for it.

**Comment by ronso0** [comment]
> The cover art toggle icon feels out of place and IMO wastes space

yeah, this can be removed. I notice I don't use it at all anymore. It was intended as a quick toggle instead of going to the View menu, so IMO it doesn't make sense to have it in Skin Settings.

The splitter graphic shows up when I load now-merged & updated Tango from lib-redesign branch. There were issues after rebasing but it should be fine now.

**Comment by ronso0** [comment]
follow-up is #1723

**Pull request merged**